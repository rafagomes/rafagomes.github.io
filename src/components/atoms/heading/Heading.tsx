import React, { ReactElement } from "react";

type Rank = "1" | "2" | "3" | "4" | "5" | "6";

interface Props {
    rank: Rank;
    children: string;
}

function Heading({ rank, children, ...otherProps }: Props): ReactElement {
    const tag = `h${rank}`;

    return React.createElement(tag, otherProps, children);
}

export default Heading;
