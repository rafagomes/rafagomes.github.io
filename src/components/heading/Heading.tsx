import React from "react"

interface Props {
    rank: string
    children: string
}

function Heading({ rank, children, ...otherProps }: Props) {
    const tag = parseInt(rank) > 6 ? "h6" : `h${rank}`

    return React.createElement(tag, otherProps, children)
}

export default Heading
