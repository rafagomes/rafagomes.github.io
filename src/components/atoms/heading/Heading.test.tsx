import React from "react";
import "../../../setupTests";
import { mount } from "enzyme";
import Heading from "./Heading";

describe("Heading", () => {
    const headingProps = { rank: "1", title: "Title" };

    it("Heading component renders text inside it", () => {
        const heading = mount(<Heading rank={headingProps.rank}>{headingProps.title}</Heading>);
        expect(heading.text()).toBe("Title");
    });

    it("Heading component render with rank correctly", () => {
        const heading = mount(<Heading rank={headingProps.rank}>{headingProps.title}</Heading>);
        expect(heading).toBe(<h1>Title</h1>);
    });
});
