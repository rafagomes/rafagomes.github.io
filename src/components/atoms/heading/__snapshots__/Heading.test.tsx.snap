// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Heading component renders as expected 1`] = `<h1 />`;
